### PR TITLE
migrate sns to new arch

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | yes
| 🔗 Related issues | #347 
| ❓ Documentation  | no

## Description

migrate
sns_topic
sns_topic_policy
sns_topic_subscription

### CODE REVIEW

File: pkg/driftctl_test.go [hunk 1]
```diff
@@ -822,51 +822,55 @@ func TestDriftctlRun_Middlewares(t *testing.T) {
...
 				},
 			},
 			mocks: func(factory resource.ResourceFactory) {
-				foo := cty.ObjectVal(map[string]cty.Value{
-					"id":     cty.StringVal("foo"),
-					"arn":    cty.StringVal("arn"),
-					"policy": cty.StringVal("{\"policy\":\"bar\"}"),
-				})
-				factory.(*terraform.MockResourceFactory).On("CreateResource", mock.MatchedBy(func(input map[string]interface{}) bool {
+				factory.(*terraform.MockResourceFactory).On("CreateAbstractResource", "foo", "aws_sns_topic_policy", mock.MatchedBy(func(input map[string]interface{}) bool {
```
Comment (eliecharra) [inline pkg/driftctl_test.go:847]: `mock.MatchedBy` can be removed and you can use directly a `map[string]interface{}`

File: pkg/middlewares/aws_sns_topic_policy_expander.go [hunk 2]
```diff
@@ -44,43 +47,35 @@ func (m AwsSNSTopicPolicyExpander) Execute(_, resourcesFromState *[]resource.Res
...
-	data := map[string]interface{}{
-		"arn":    topic.Arn,
-		"id":     topic.Id,
-		"policy": topic.Policy,
-	}
-	ctyVal, err := m.resourceFactory.CreateResource(data, "aws_sns_topic_policy")
-	if err != nil {
-		return err
+	arn, exist := topic.Attrs.Get("arn")
+	if !exist || arn == "" {
```
Comment (eliecharra) [inline pkg/middlewares/aws_sns_topic_policy_expander.go:57]: This case was not handled previously, can it really happen ?

File: pkg/middlewares/aws_sns_topic_policy_expander_test.go [hunk 3]
```diff
@@ -27,83 +27,198 @@ func TestAwsSNSTopicPolicyExpander_Execute(t *testing.T) {
...
+						"arn":    "arn",
+						"id":     "ID",
+						"policy": "{\"policy\":\"coucou\"}",
+					},
 				},
 			},
 			mock: func(factory *terraform.MockResourceFactory) {
-				factory.On("CreateResource", mock.Anything, "aws_sns_topic_policy").Once().Return(nil, nil)
+				factory.On("CreateAbstractResource", "ID", awsresource.AwsSnsTopicPolicyResourceType, mock.MatchedBy(func(input map[string]interface{}) bool {
+					return input["id"] == "ID"
```
Comment (eliecharra) [inline pkg/middlewares/aws_sns_topic_policy_expander_test.go:61]: Same remark as in another PR, we should change the ID's or either change the matcher

Comment (moadibfr) [reply pkg/middlewares/aws_sns_topic_policy_expander_test.go:61]: I don't get this remark ? There is only one resource so only this id ?

File: pkg/middlewares/aws_sns_topic_policy_expander.go (hunk 2 above)
Comment (eliecharra) [reply pkg/middlewares/aws_sns_topic_policy_expander.go:57]: Yeah it can't hurt it was just to be sure to understand.

**Pull request merged**